src/app/pages/categories/management/pages-categories-management.component.spec.ts

```diff
@@ -0,0 +1,168 @@
+        });
+
+        describe('not empty table', () => {
+          it('should display categories in table', fakeAsync(() => {
+            categoriesSubject.next(categoriesObjectsMock);
+            flushMicrotasks();
+            fixture.detectChanges();
+
+            const table = fixture.debugElement.query(
+              By.css('.mat-table'),
+            ).nativeElement as HTMLDivElement;
+
+            expect(table).toBeTruthy();
+            expect(component.transactionCategoriesData.hasError).toBeFalse();
+            expect(component.transactionCategoriesData.data).toEqual(categoriesObjectsMock);
+          }));
+        });
+      });
+
+      describe('error', () => {
+        it('should set TStateData.error as true and data as null', fakeAsync(() => {
+          categoriesSubject.error('Some error');
+          flushMicrotasks();
+
+          expect(component.transactionCategoriesData.hasError).toBeTrue();
+          expect(component.transactionCategoriesData.data).toEqual(null);
+        }));
+
+        it('should display error wall', fakeAsync(() => {
+          categoriesSubject.error('Some error');
+          flushMicrotasks();
+          fixture.detectChanges();
+
+          const messageDiv = fixture.debugElement.query(
+            By.css('.message-container'),
+          ).nativeElement as HTMLDivElement;
+
+          expect(messageDiv).toBeTruthy();
+        }));
+      });
+    });
+  });
+
+  describe('filter transactions', () => {
+    let matSelect: HTMLDivElement;
+
+    beforeEach( fakeAsync(() => {
+      categoriesSubject.next(categoriesObjectsMock);
+      flushMicrotasks();
+      fixture.detectChanges();
+
+      matSelect = fixture.debugElement.query(
+        By.css('.mat-select-trigger'),
+      ).nativeElement as HTMLDivElement;
+
+      matSelect.click();
+      fixture.detectChanges();
+    }));
+
+    describe('filter categories by Expenses', () => {
+      it('selected Expenses on select-box should filter categories with expense type', fakeAsync( async () => {
+        const expensesOption: HTMLDivElement = fixture.debugElement.queryAll(
+          By.css('.mat-option'),
+        )[1].nativeElement as HTMLDivElement;
+        expensesOption.click();
+        fixture.detectChanges();
+        flushMicrotasks();
+        flush();
+
+        await fixture.whenStable();
+
+        expect(component.displayedCategories.length)
+          .toBe(categoriesObjectsMock.filter(c => c.type === WalletTransactionType.Expense).length);
+      }));
+    });
+
+    describe('filter categories by Incomes', () => {
+      it('selected Incomes on select-box should filter categories with incomes type', fakeAsync( async () => {
+        const incomesOption: HTMLElement = fixture.debugElement.queryAll(
+          By.css('.mat-option'),
+        )[2].nativeElement as HTMLDivElement;
+
+        incomesOption.click();
+        fixture.detectChanges();
+        flushMicrotasks();
+        flush();
+        // test error
```

Comment:
This is no longer failing for me but you forgot to add to the declarations here so there is a warning in the terminal when running tests:
```
RROR: 'NG0304: 'app-no-results' is not a known element (used in the 'PagesCategoriesManagementComponent' component template):
1. If 'app-no-results' is an Angular component, then verify that it is a part of an @NgModule where this component is declared.
2. If 'app-no-results' is a Web Component then add 'CUSTOM_ELEMENTS_SCHEMA' to the '@NgModule.schemas' of this component to suppress this message.'
```